Automattic/jetpack
# AI Assistant: Ignore first chunk on unclear prompt check

Fixes an error where, somehow, the first response chunk is either not sent, received or generated and the prompt was not undestood, ending up with this content:

![image](https://github.com/Automattic/jetpack/assets/8486249/013594f1-b654-43d3-afee-a73f52b477b9)

## Proposed changes:

* Refactors the logic a bit to make it more easily understood
* Ignores the first `__` in the message when checking for the error marker
* Tweaks and moves the prompt

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






This one is very tricky to test because I was not able to reproduce the problem normally.
* Add an AI Assistant block and ask some nonsense like "sdjsd"
* Check that the notice about trying again is displayed as normal
* Check that subsequent requests, both clear and unclear, are treated correctly

To simulate a state where the first `__` is not received, it would be possible with something like this:
```js
		if ( chunk && chunk !== '__' ) {
			this.fullMessage += chunk;
```
